go/vt/dbconfigs/dbconfigs.go

```diff
@@ -237,27 +237,23 @@ func HasConnectionParams() bool {
 // is used to initialize the per-user conn params.
 func Init(defaultSocketFile string) (*DBConfigs, error) {
 	// The new base configs, if set, supersede legacy settings.
-	if HasConnectionParams() {
-		for _, uc := range dbConfigs.userConfigs {
+	for _, uc := range dbConfigs.userConfigs {
+		if HasConnectionParams() {
 			uc.param.Host = baseConfig.Host
 			uc.param.Port = baseConfig.Port
 			uc.param.UnixSocket = baseConfig.UnixSocket
-			uc.param.Charset = baseConfig.Charset
-			uc.param.Flags = baseConfig.Flags
-			if uc.useSSL {
-				uc.param.SslCa = baseConfig.SslCa
-				uc.param.SslCaPath = baseConfig.SslCaPath
-				uc.param.SslCert = baseConfig.SslCert
-				uc.param.SslKey = baseConfig.SslKey
-				uc.param.ServerName = baseConfig.ServerName
-			}
+		} else {
+			uc.param.UnixSocket = defaultSocketFile
 		}
-	} else {
-		// Use supplied socket value if conn parameters are not specified.
-		for _, uc := range dbConfigs.userConfigs {
-			if uc.param.UnixSocket == "" && uc.param.Host == "" {
-				uc.param.UnixSocket = defaultSocketFile
-			}
+
+		uc.param.Charset = baseConfig.Charset
+		uc.param.Flags = baseConfig.Flags
+		if uc.useSSL {
```

Comment:
I'm not sure whether these should be left out when using a socket. @sougou what do you think?

Reply:
It's not very useful to use SSL for sockets, but we don't care.